--- orig/rawdog-2.12/rawdoglib/rawdog.py	2009-02-06 17:41:58.000000000 +0000
+++ rawdog-2.12/rawdoglib/rawdog.py	2010-07-21 15:08:47.659987002 +0100
@@ -435,6 +435,10 @@
 		sequence = 0
 		for entry_info in p["entries"]:
 			article = Article(feed, entry_info, now, sequence)
+			for feedconfig in config["feedslist"]:
+				if feedconfig[0] == feed:
+					if feedconfig[2].has_key("define_microblog") and feedconfig[2]["define_microblog"] == "true":
+						article.twitter = True
 			ignore = plugins.Box(False)
 			plugins.call_hook("article_seen", rawdog, config, article, ignore)
 			if ignore.value:
@@ -502,6 +506,7 @@
 		self.feed = feed
 		self.entry_info = entry_info
 		self.sequence = sequence
+		self.twitter = False
 
 		modified = entry_info.get("modified_parsed")
 		self.date = None
@@ -1322,7 +1327,23 @@
 			else:
 				title = "Link"
 
-		itembits["title_no_link"] = title
+		if 'twitter' in dir(article) and article.twitter:
+			split = title.split(":", 1)
+			text = split[0] + "</a>"
+			split[1] = split[1].lstrip()
+			if split[1].startswith("@"):
+				atSplit = split[1].split(" ", 1)
+				twitterLink = atSplit[0].lstrip("@")
+				twitterLink = twitterLink.rstrip(":")
+                                if len(atSplit) >= 2:
+				    text = text + " <a href='http://twitter.com/"+ twitterLink + "'>" + atSplit[0] + "</a>" + " " + atSplit[1]
+                                else:
+				    text = text + " <a href='http://twitter.com/"+ twitterLink + "'>" + atSplit[0] + "</a>"
+			else:
+				text = text + " " + split[1]
+			itembits["title_no_link"] = text
+		else:
+			itembits["title_no_link"] = title
 		if link is not None:
 			itembits["url"] = string_to_html(link, config)
 		else:
@@ -1361,7 +1382,10 @@
 			itembits["date"] = ""
 
 		plugins.call_hook("output_item_bits", self, config, feed, article, itembits)
-		itemtemplate = self.get_itemtemplate(config)
+                if 'twitter' in dir(article) and article.twitter:
+			itemtemplate = load_file("microblogitemtemplate")
+		else:
+			itemtemplate = self.get_itemtemplate(config)
 		f.write(fill_template(itemtemplate, itembits))
 
 	def write_remove_dups(self, articles, config, now):
@@ -1445,7 +1469,7 @@
 
 		return bits
 
-	def write_output_file(self, articles, article_dates, config):
+	def write_output_file(self, articles, twitterArticles, article_dates, config):
 		"""Write a regular rawdog HTML output file."""
 		f = StringIO()
 		dw = DayWriter(f, config)
@@ -1464,6 +1488,19 @@
 		bits["items"] = f.getvalue()
 		f.close()
 		bits["num_items"] = str(len(articles))
+
+                #TWITTER
+		f = StringIO()
+		dw = DayWriter(f, config)
+
+		for article in twitterArticles[:60]:
+			self.write_article(f, article, config)
+
+		dw.close()
+
+		bits["twitter"] = f.getvalue()
+                #end of TWITTER
+
 		plugins.call_hook("output_bits", self, config, bits)
 		s = fill_template(self.get_template(config), bits)
 		outputfile = config["outputfile"]
@@ -1530,6 +1567,15 @@
 				articles.append(a)
 				article_dates[a] = -date
 
+		twitterArticles = []
+		normalArticles = []
+		for article in articles:
+			if 'twitter' in dir(article) and article.twitter == True:
+				twitterArticles.append(article)
+			else:
+				normalArticles.append(article)
+		articles = normalArticles
+
 		plugins.call_hook("output_write", self, config, articles)
 
 		if not plugins.call_hook("output_sorted_filter", self, config, articles):
@@ -1540,7 +1586,7 @@
 		config.log("Selected ", len(articles), " of ", numarticles, " articles to write; ignored ", dup_count, " duplicates")
 
 		if not plugins.call_hook("output_write_files", self, config, articles, article_dates):
-			self.write_output_file(articles, article_dates, config)
+			self.write_output_file(articles, twitterArticles, article_dates, config)
 
 		config.log("Finished write")
 
