--- planetkde/plugins/rss.py~   2010-08-13 11:00:45.506801003 +0100
+++ planetkde/plugins/rss.py    2010-08-16 09:42:49.582541002 +0100
@@ -162,8 +153,19 @@
         except ValueError:
             maxarticles = len(articles)
         for article in articles[:maxarticles]:
-            xml_article = channel.newChild(None, 'item', None)
-            self.article_to_xml(xml_article, rawdog, config, article)
+            #Planet KDE addition, don't include articles in a feedclass
+            feed = rawdog.feeds[article.feed]
+            itembits = {}
+            toAdd = True;
+            for name, value in feed.args.items():
+                if name.startswith("define_"):
+                    itembits[name[7:]] = value
+                    if "feedclass" in itembits:
+                        toAdd = False;
+            
+            if toAdd: 
+                xml_article = channel.newChild(None, 'item', None)
+                self.article_to_xml(xml_article, rawdog, config, article)
 
         doc.saveFormatFile(self.options["outputxml"], 1)
         doc.freeDoc()
