--- planetkde/plugins/rss.py~	2010-08-13 11:00:45.506801003 +0100
+++ planetkde/plugins/rss.py	2010-08-16 09:42:49.582541002 +0100
@@ -105,6 +101,15 @@
             return feed.get_html_name(config)
 
     def article_to_xml(self, xml_article, rawdog, config, article):
+        #Planet KDE addition, don't include articles in a feedclass
+        feed = rawdog.feeds[article.feed]
+        itembits = {}
+	for name, value in feed.args.items():
+	    if name.startswith("define_"):
+	        itembits[name[7:]] = value
+        if "feedclass" in itembits:
+            return
+
         entry_info = article.entry_info
 
         id = entry_info.get("id", self.options["xmlurl"] + "#id" + article.hash)
