Index: planetkde/plugins/rss.py
===================================================================
--- planet/planetkde/plugins/rss.py	(revision 1210312)
+++ planet/planetkde/plugins/rss.py	(working copy)
@@ -110,9 +110,12 @@
         title = escape(self.feed_name(rawdog.feeds[article.feed], config))
         s = detail_to_html(entry_info.get("title_detail"), True, config)
         if s is not None:
-            title += ": " + s.encode('utf8')
+            title = s.encode('utf8')
         xml_article.newChild(None, 'title', title)
 
+        author = escape(self.feed_name(rawdog.feeds[article.feed], config))
+        xml_article.newChild(None, 'author', author)
+
         if article.date is not None:
             date = rfc822_date(gmtime(article.date))
             xml_article.newChild(None, 'pubDate', date)
