--- ../../planet/rawdoglib/rawdog.py	2010-12-30 17:26:46.265332000 +0000
+++ rawdoglib/rawdog.py	2010-12-30 17:25:34.593332001 +0000
@@ -1390,6 +1390,9 @@
 			itemtemplate = load_file("microblogitemtemplate")
 		else:
 			itemtemplate = self.get_itemtemplate(config)
+
+		itembits['name'] = unicode(itembits['name'], 'utf-8').encode("ascii", "xmlcharrefreplace")
+
 		f.write(fill_template(itemtemplate, itembits))
 
 	def write_remove_dups(self, articles, config, now):
